File: LeetCode/338. Counting Bits.cpp
// LeetCode 338. Counting Bits
// Brian Kernighan Algorithm
class Solution {
public:
    int countSetBits(int n){
        int count = 0;
        while(n)
            n &= (n - 1), count++;
        return count;
    }
    vector<int> countBits(int n) {
        vector<int> vr;
        for(int i = 0; i <= n; i++)
            vr.push_back(countSetBits(i));
        return vr;
    }
};

File: LeetCode/338. Counting Bits.py
# LeetCode 338. Counting Bits
# Brian Kernighan Algorithm
class Solution:
    def setBits(self, n : int) -> int : 
        count = 0
        while n :
            n &= (n - 1)
            count += 1
        return count

    def countBits(self, n: int) -> List[int]:
        arr = []
        for i in range(0, n + 1) : 
            arr.append(self.setBits(i))
        return arr

# Snek method
class Solution:
    def countBits(self, n: int) -> List[int]:
        arr = []
        for i in range(0, n + 1) :
            arr.append(i.bit_count())
        return arr